nova/tests/unit/scheduler/weights/test_weights_ram.py

```diff
@@ -109,3 +113,50 @@ def test_ram_filter_negative(self):
         weighed_host = weights[-1]
         self.assertEqual(0, weighed_host.weight)
         self.assertEqual('negative', weighed_host.obj.host)
+
+
+class RamResizeBigVmToSameHostWeigherTestCase(test.NoDBTestCase):
+    def setUp(self):
+        super(RamResizeBigVmToSameHostWeigherTestCase, self).setUp()
+        self.weight_handler = weights.HostWeightHandler()
+        self.weighers = [ram.RamResizeBigVmToSameHostWeigher()]
+
+    def _get_hosts_and_request_spec(self):
+        # simulate a resizing instance
+        instances = [
+            objects.Instance(
+                uuid='b1950038-1a9b-48ac-8cb9-9584955dec47',
+                old_flavor=objects.Flavor(memory_mb=CONF.bigvm_mb,
+                                          extra_specs={}),
+                new_flavor=objects.Flavor(memory_mb=CONF.bigvm_mb/2,
+                                          extra_specs={}),
+            )
+        ]
+        hosts = [
+            fakes.FakeHostState('same_host', 'n1', {}, instances),
+            fakes.FakeHostState('other_host', 'n2', {}),
+        ]
+        spec_obj = objects.RequestSpec(instance_uuid=instances[0].uuid)
+        return hosts, spec_obj
+
+    def test_ram_resize_to_same_host(self):
+        self.flags(ram_resize_to_same_host_weight_multiplier=1.0,
+                   group='filter_scheduler')
+        hosts, spec_obj = self._get_hosts_and_request_spec()
+        weighed_hosts = self.weight_handler.get_weighed_objects(self.weighers,
+                                                                hosts,
+                                                                spec_obj)
+        self.assertEqual(1.0, weighed_hosts[0].weight)
+        self.assertEqual(0.0, weighed_hosts[1].weight)
+        self.assertEqual('same_host', weighed_hosts[0].obj.host)
+
+    def test_ram_resize_to_different_host(self):
+        self.flags(ram_resize_to_same_host_weight_multiplier=-1.0,
+                   group='filter_scheduler')
+        hosts, spec_obj = self._get_hosts_and_request_spec()
+        weighed_hosts = self.weight_handler.get_weighed_objects(self.weighers,
+                                                                hosts,
+                                                                spec_obj)
+        self.assertEqual(0.0, weighed_hosts[0].weight)
+        self.assertEqual(-1.0, weighed_hosts[1].weight)
+        self.assertEqual('other_host', weighed_hosts[0].obj.host)
```

Comment:
There are tests missing for
1. normal scheduling
2. migration (basically resize without flavor change)
3. rebuild?
4. unshelve?

Reply:
1. Instance doesn't exist and cannot be on any host. Returns 0.0 in `resize_same_host.py` L66:
```python
        if request_spec.instance_uuid not in host_state.instances:
            return 0.0
```
2. All flavor attributes should be the same. If an migration to a specific host is requested this weigher should not be asked.
3. rebuild -- use utils.`request_is_rebuild()`
4. see 1.

Reply:
1. The test is sill missing. You've got a test for "non-resizing", but not for "not on any host". Technically, there's test because of one the host doesn't have the instance, but please make it explicit.
4. see 1, if at all possible to specify this. E.g. your instances lack a host attribute. I know you don't use it, but theoretically it might be used and your test should also work for other implementations done in the future.

Reply:
1. Done
2. I don't really understand this point. I added a 'host' attr -- all for coding for change -- but I don't know how this releates to "see 1". There is a non-resizing, a.k.a migrating instance. Would you prefer to have the test renamed?

Reply:
I'd like to have a test for migration (i.e. host is set and it's in instances of a host) and I'd like to have a test for initial scheduling (i.e. host is unset and it's not in instances of any host).

Reply:
Then that's the case: `test_ignore_non_resizing_instance` and `test_ignore_scheduling_new_instance`. If you think it's better I can rename the "non_resizing" to "migrating".